GVFS/GVFS.Service/ProductUpgradeTimer.cs

```diff
@@ -23,12 +23,19 @@ public void Start()
             Random random = new Random();
             TimeSpan startTime = TimeSpan.Zero;
 
-            this.tracer.RelatedInfo("Starting auto upgrade checks.");
-            this.timer = new Timer(
-                this.TimerCallback,
-                state: null,
-                dueTime: startTime,
-                period: TimeInterval);
+            if (!ProcessHelper.IsDevelopmentVersion())

```

Comment:
As we've discussed in past PRs, I don't think you should ever disable a feature based on it being a development version. It is incredibly risky to ship code to customers that is different than the code we run ourselves.

And in this case, this check will not prevent the failures anyway, at least not on dev boxes. If I am dogfooding the product and have a non-dev version installed and running, it'll happily download the next upgrade. Then when I build and run the tests on my dev version, it will find those downloads. Yes, you're currently deleting them on test startup, but that is racy as the download could happen right after it.

Reply:
Alternate approach is to look for upgrade messaging in the git output and ignore it in the Functional Tests. Have incorporated this, please take a look.